test/promises-async-await.spec.js

```diff
@@ -0,0 +1,66 @@
+import assert from "assert";
+
+describe("Promises", function () {
```

Comment:
On promises, one more case to cover is how to handle errors in an async function.

Reply:
added error handling for async functions